src/agent/src/windows_service.cpp

```diff
@@ -0,0 +1,176 @@
+#include <windows_service.hpp>
+
+#include <logger.hpp>
+
+#include <chrono>
+#include <thread>
+
+SERVICE_STATUS g_ServiceStatus = {};
+SERVICE_STATUS_HANDLE g_StatusHandle = nullptr;
+HANDLE g_ServiceStopEvent = INVALID_HANDLE_VALUE;
+
+namespace
+{
+    const std::string AGENT_SERVICENAME = "Wazuh Agent";
+
+    std::string GetExecutablePath()
+    {
+        char buffer[MAX_PATH];
+        GetModuleFileName(NULL, buffer, MAX_PATH);
+        return std::string(buffer);
+    }
+} // namespace
+
+namespace WindowsService
+{
+
+    void InstallService()
+    {
+        const std::string exePath = GetExecutablePath() + " service";
+
+        SC_HANDLE schSCManager = OpenSCManager(nullptr, nullptr, SC_MANAGER_CREATE_SERVICE);
+        if (schSCManager == nullptr)
```

Comment:
There's a mix of styles when checking pointers in these files, sometimes comparing explicitly against `nullptr` and sometimes checking the implicit conversion to bool. Can we settle for the latter? (either one is ok...)

Reply:
Fixed! I decided to use `if (!varName)`.